minio/api.py

```diff
@@ -1546,14 +1554,28 @@ def _stream_put_object(self, bucket_name, object_name,
         if total_uploaded != content_size:
             msg = 'Data uploaded {0} is not equal input size ' \
                   '{1}'.format(total_uploaded, content_size)
+            # cleanup incomplete upload upon incorrect upload
+            # automatically
+            self._remove_incomplete_upload(bucket_name,
+                                           object_name,
+                                           upload_id)
             raise InvalidSizeError(msg)
 
         # Complete all multipart transactions if possible.
-        mpart_result = self._complete_multipart_upload(bucket_name,
-                                                       object_name,
-                                                       upload_id,
-                                                       uploaded_parts,
-                                                       metadata=metadata)
+        try:
```

Comment:
Looks like there is formatting issue here

Reply:
why do you think?

Reply:
i don't see any issue, if there is a formatting issue python won't really compile.